alica_engine/src/engine/PlanPool.cpp

```diff
@@ -0,0 +1,66 @@
+#include "engine/AlicaEngine.h"
+#include "engine/IPlanCreator.h"
+#include "engine/PlanPool.h"
+#include "engine/RunningPlan.h"
+#include "engine/BasicPlan.h"
+#include "engine/PlanRepository.h"
+#include "engine/model/Configuration.h"
+#include "engine/model/ConfAbstractPlanWrapper.h"
+#include "engine/model/Plan.h"
+
+namespace alica
+{
+
+PlanPool::PlanPool(AlicaEngine* ae)
+        : _ae(ae)
+{
+}
+
+PlanPool::~PlanPool() = default;
+
+bool PlanPool::init(IPlanCreator& planCreator)
+{
+    const PlanRepository::Accessor<ConfAbstractPlanWrapper>& wrappers =
+            _ae->getPlanRepository().getConfAbstractPlanWrappers();
+    for (const ConfAbstractPlanWrapper* wrapper : wrappers) {
+        if (const auto* plan = dynamic_cast<const Plan*>(wrapper->getAbstractPlan())) {
+            if (getBasicPlan(plan, wrapper->getConfiguration())) {
```

Comment:
Remember that `getBasicPlan` above creates a `shared_ptr`? So do not use it for a check, use a separate `planExists` method. (or remove the `shared_ptr`).